targets/TARGET_NORDIC/TARGET_NRF5x/mbed_lib.json

```diff
@@ -0,0 +1,29 @@
+{
+    "name": "nordic",
+    "config": {
+        "uart-hwfc": {
```

Comment:
@marcuschangarm @0xc0170  Would having uart declared in both mbed_lib.json and targets.json cause any issues when overriding it in custom_targets.json?

I'm using a custom_targets.json which is intended to disable flow control, but the mbed_config.h always ends up containing the following:

```
#define MBED_CONF_NORDIC_UART_HWFC                        1    // set by target:MCU_NRF52840
#define MBED_CONF_NORDIC_UART_HWFC                        1    // set by library:nordic
```
Since I'm not using hardware flow control this results in my printf's hanging.

Reply:
I think you might have missed my response here: https://os.mbed.com/forum/upcoming-features/topic/29477/?page=1#comment-56395

In any case, targets.json has been cleaned up and a new PR is in the pipeline: https://github.com/ARMmbed/mbed-os/pull/6547